core/cli_helpers.py

```diff
@@ -349,6 +349,9 @@ def command_catalog(self):
             'ceph_volume_lvm_list':
                 [BinCmd('ceph-volume lvm list'),
                  FileCmd('sos_commands/ceph/ceph-volume_lvm_list')],
+            'ceph_report_json_decoded':
+                [BinCmd('ceph report', json_decode=True),
+                 FileCmd('sos_commands/ceph/ceph_report')],
```

Comment:
I assume you are not doing json_decode=True for both because the sosreport one has the spurious data at the end of the file. Because of that I think we should create a custom FileCmd like:

class CephReportFileCmd(FileCmd):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.register_hook("pre-exec", self.sanitise_json_contents)

    def sanitise_json_contents(self, **kwargs):
        ...

And move the code that does the fix for the last line in the file to here.